storage/database/db_manager.go

```diff
@@ -1057,30 +1153,106 @@ func (dbm *databaseManager) WriteMerkleProof(key, value []byte) {
 
 // Cached Trie Node operation.
 func (dbm *databaseManager) ReadCachedTrieNode(hash common.Hash) ([]byte, error) {
+	if dbm.inMigration {
+		if val, err := dbm.newStateTrieDB.Get(hash[:]); err == nil {
+			return val, nil
+		}
+	}
+	return dbm.ReadCachedTrieNodeFromOld(hash)
+}
+
+// Cached Trie Node Preimage operation.
+func (dbm *databaseManager) ReadCachedTrieNodePreimage(secureKey []byte) ([]byte, error) {
+	if dbm.inMigration {
+		if val, err := dbm.newStateTrieDB.Get(secureKey); err == nil {
+			return val, nil
+		}
+	}
+	return dbm.ReadCachedTrieNodePreimageFromOld(secureKey)
+}
+
+// State Trie Related operations.
+func (dbm *databaseManager) ReadStateTrieNode(key []byte) ([]byte, error) {
+	if dbm.inMigration {
+		if val, err := dbm.newStateTrieDB.Get(key); err == nil {
+			return val, nil
+		}
+	}
+	return dbm.ReadStateTrieNodeFromOld(key)
+}
+
+func (dbm *databaseManager) HasStateTrieNode(key []byte) (bool, error) {
+	val, err := dbm.ReadStateTrieNode(key)
+	if val == nil || err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// ReadPreimage retrieves a single preimage of the provided hash.
+func (dbm *databaseManager) ReadPreimage(hash common.Hash) []byte {
+	if dbm.inMigration {
+		if val, err := dbm.newStateTrieDB.Get(preimageKey(hash)); err == nil {
+			return val
+		}
+	}
+	return dbm.ReadPreimageFromOld(hash)
+}
+
+func (dbm *databaseManager) ReadCachedTrieNodeFromOld(hash common.Hash) ([]byte, error) {
 	db := dbm.getDatabase(StateTrieDB)
 	return db.Get(hash[:])
 }
 
 // Cached Trie Node Preimage operation.
-func (dbm *databaseManager) ReadCachedTrieNodePreimage(secureKey []byte) ([]byte, error) {
+func (dbm *databaseManager) ReadCachedTrieNodePreimageFromOld(secureKey []byte) ([]byte, error) {
 	db := dbm.getDatabase(StateTrieDB)
 	return db.Get(secureKey)
 }
 
 // State Trie Related operations.
-func (dbm *databaseManager) ReadStateTrieNode(key []byte) ([]byte, error) {
+func (dbm *databaseManager) ReadStateTrieNodeFromOld(key []byte) ([]byte, error) {
 	db := dbm.getDatabase(StateTrieDB)
 	return db.Get(key)
 }
 
-func (dbm *databaseManager) HasStateTrieNode(key []byte) (bool, error) {
-	val, err := dbm.ReadStateTrieNode(key)
+func (dbm *databaseManager) HasStateTrieNodeFromOld(key []byte) (bool, error) {
+	db := dbm.getDatabase(StateTrieDB)
+	val, err := db.Get(key)
 	if val == nil || err != nil {
 		return false, err
 	}
 	return true, nil
 }
 
+// ReadPreimage retrieves a single preimage of the provided hash.
+func (dbm *databaseManager) ReadPreimageFromOld(hash common.Hash) []byte {
+	if dbm.inMigration {
```

Comment:
The code under this condition is not necessary because it runs in `ReadPreimage`.

Reply:
Right! I forgot to remove this lines. Thanks!